sub/sd_ass.c

```diff
@@ -289,45 +289,36 @@ static int init(struct sd *sd)
 }
 
 // Check if subtitle has events that would cause it to be animated inside {}
-static bool is_animated(char *s)
+static bool is_animated(const char *str)
 {
-    bool in_tag = false;
-    bool valid_event = false;
-    bool valid_tag = false;
-    while (*s) {
-        if (!in_tag && s[0] == '{')
-            in_tag = true;
-        if (s[0] == '\\') {
-            s++;
-            if (!s[0])
-                break;
-            if (s[0] == 'k' || s[0] == 'K' || s[0] == 't') {
-                valid_event = true;
-                continue;
-            // just bruteforce the multi-letter ones
-            } else if (s[0] == 'f') {
-                if (!strncmp(s, "fad", 3)) {
-                    valid_event = true;
-                    continue;
-                }
-            } else if (s[0] == 'm') {
-                if (!strncmp(s, "move", 4)) {
-                    valid_event = true;
-                    continue;
-                }
+    if (str[0] == '{')
+        goto found;
+    while ((str = strchr(str, '{'))) {
+        ++str;
+        if (str[-2] == '\\')
+            continue;
+found:;
```

Comment:
This seems overly complex and fragile; why not just do a the `if (str > begin && str[-1] == '\\')` like I'd suggested?

Reply:
> complex and fragile

Why?

> why not just do a the if (str > begin && str[-1] == '\\') like I'd suggested?

Because we don't need to check this, except first iteration. Saves registers, and instructions.

Reply:
This isn't in the hot part of the function, and the difference is negligible; I don't think it's worth the more opaque code.

If you do keep it, at least move `++str` to after `found:` and use `str[-1] == '\\'` for the check. Alternately, use `*str++ == '{'` for the initial check, and move the `++str` to immediately before `found:`.

Reply:
I don't mind, it is insignificant, changed.